### PR TITLE
Rebase develop after a master push

### PR DESCRIPTION
**Pull request by skullydazed**
## Description

Should keep develop more up to date with master and reduce maintenance burden.

## Types of Changes

- [x] Core
- [x] Enhancement/optimization

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by drashna** [comment]
Is there a way to ensure that this only happens on the qmk repo?

**Comment by skullydazed** [comment]
> Is there a way to ensure that this only happens on the qmk repo?

a46e508da adds an if statement so it only happens for the qmk repo.

**Review by noroadsleft — APPROVED** [review]
`__attribute__ ((weak))` :heavy_check_mark:

Not well-versed in Workflows, but looks okay to my understanding.

**Comment by tzarc** [comment]
Whilst this may help keep `develop` up to date and lower our own maintenance effort, it's likely this will have a profoundly negative effect on anyone attempting to PR against `develop`.

I think we should really consider going down the merge route instead of rebase, at least in the context of `master`<->`develop`.

**Comment by drashna** [comment]
> I think we should really consider going down the merge route instead of rebase, at least in the context of `master`<->`develop`.

I agree.  It's already an issue, as is.  But having this done automatically means that ANY pr that is opened against develop will have a large burden placed on contributors. 

Since we squash develop into master periodically, the merge commits shouldn't be an issue.

**Comment by tzarc** [comment]
I thought develop -> master was a merge commit, not a squash, so that we keep history?

Either way, my point is that for master to and from develop, we should have merges instead of squashes in general.

**Comment by skullydazed** [comment]
> I thought develop -> master was a merge commit, not a squash, so that we keep history?

So far we've always done rebases, but yes the end result is the same.

When it comes to the topic of merge commits, and how much work we should go through to avoid them, I'm reminded of how I felt when I started working with git in 2008- what's the big deal? I didn't understand why some people cared so much, but over time I developed habits to make them happy. Now it's something I do out of reflex. If general sentiment has changed over time so that merge commits aren't seen as a bad thing I'm happy to switch to a merge strategy. The part I care about most here is that commits from master get to develop as rapidly as possible.

Before I ring that bell does anyone want to make the case for rebasing and avoiding merge commits?

File: .github/workflows/develop_update.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        fetch-depth: 0
+
+    - name: Update develop from master
+      shell: 'bash {0}'
+      run: |
+        if [ -e .locked ]; then
+            echo 'Develop branch is locked, aborting update!'
+            exit 1
+        fi
+
```
Comment (zvecr) [inline .github/workflows/develop_update.yml:27]: Something like this maybe?
```suggestion
    - name: Check development not locked
      run: |
        test-f .locked

    - name: Update develop from master
      run: |
```

**Review by zvecr — CHANGES_REQUESTED** [review]
Is the intention for the `.locked` file exists on master or develop? The process might have to be slightly different depending on the intention, as `actions/checkout` checks out master right now.

**Comment by tzarc** [comment]
> Before I ring that bell does anyone want to make the case for rebasing and avoiding merge commits?

If I'm brutally honest, I would like rebasing instead of merges... if it were just people who are familiar with git making PRs. 

Observing our contributors however, it seems that there's a general lack in familiarity with the processes involved in doing so. The friction that rebases create, coupled with the rate of PR merges in general, results in frustration for both parties.

I guess before we finalise this decision -- do we accept that there's a bit more work for us managing these branches in order to minimise contributor frustration, or do we expect a certain level of experience with the usage of git, and enforce rebasing as the way forward?

File: .github/workflows/develop_update.yml (hunk 1 above)
Comment (skullydazed) [reply .github/workflows/develop_update.yml]: I rearranged it into something that looks more like what you have. How's that look?

**Comment by drashna** [comment]
> I guess before we finalise this decision -- do we accept that there's a bit more work for us managing these branches in order to minimise contributor frustration, or do we expect a certain level of experience with the usage of git, and enforce rebasing as the way forward?

As somebody that had zero git experience when I started with QMK .... merging is easier, if messier.  And simpler.   I say keep the bar low, so that people with more passion than skill can continue to contribute (like myself 😆).  

Yes, that may put more work on us, long term. But I think it will help keep people interested in contributing, too.

**Review by tzarc — APPROVED** [review]

**Pull request merged**